scripts/source-maintenance.sh

```diff
@@ -33,8 +33,16 @@ KeepGoingDirective=""
 # has changed in different releases.
 # if --with-astyle /path/to/astyle is used, the check is still performed
 # and a warning is printed, but the sources are reformatted
+# --changed-only applies the formatting only to files changes since
+#    the fork point. Detection may fail, in this case it's applied to
+#    all files
+#
+# behaviouyr can be altered using environment variables:
+# ASTYLE: path to the astyle binary to use
+# MD5: path to the md5sum or similar checksum calculator to use
```

Comment:
I am not against supporting an environment variable that can be used to specify `source-maintenance.sh` options. The script will then use that variable to set the default values for command line options (for example). `SQUID_SOURCE_MAINTENANCE_OPTIONS`?

That would still be adding another configuration vector, but in a mostly safe way that does not cause conflicts with the existing options.

Reply:
Command-line options still take priority over environment variables-vectored configs. The order of preference is always
command-line > envvar > hardcoded

This is consistent with common idioms and satisfies the minimum-surprise principle

Reply:
I agree with your last comment but do not see the relevance.

Reply:
> I agree with your last comment but do not see the relevance.

It is a more commonly expected behaiour than any alternative, therefore it's a better choice.
If it can help bring things forward, I can also state it's a behaviour I like better :)

Reply:
> It is a more commonly expected behaiour than any alternative, therefore it's a better choice.

With the exception of well-known shell-wide standard variables like `EDITOR`, supplying program names via dedicated environment variables is probably less common than supplying them via command line options. And that "standard shell variable" exception is not applicable here.